### PR TITLE
WIP Inject html with id value instead of className

### PR DESCRIPTION
**Pull request by JeremyTX**

### CODE REVIEW

File: src/client/login-button.ts [hunk 1]
```diff
@@ -37,34 +37,33 @@ interface QueryParams {
...
+      if (clickTargetHasID(idValue, e)) {
         callback(e)
       }
     }
     const useCapture = true
     document.addEventListener("click", handleClick, useCapture)
   }
 
-  interruptClick("header-button-login", e => {
+  interruptClick("header--button-login", e => {
```
Comment (JeremyTX) [inline src/client/login-button.ts]: https://github.com/verdaccio/verdaccio/blob/v4.1.0/test/e2e/e2e.spec.js#L20

File: src/client/login-button.ts [hunk 2]
```diff
@@ -37,34 +37,33 @@ interface QueryParams {
...
     e.preventDefault()
     e.stopPropagation()
     location.href = "/-/oauth/authorize"
   })
 
-  interruptClick("header-button-logout", e => {
+  interruptClick("header--button-logout", e => {
```
Comment (JeremyTX) [inline src/client/login-button.ts]: https://github.com/verdaccio/verdaccio/blob/v4.1.0/test/e2e/e2e.spec.js#L105

**Comment by JeremyTX** [comment]
@n4bb12 please review when you get a chance.

**Comment by ian-rudge** [comment]
This is a breaking change, I think ideally it would be backwards compatible with Verdaccio v3

**Comment by seesemichaelj** [comment]
@n4bb12 I'm curious what is WIP about this PR? What needs to change for it to be merged? The current logic is broken for Verdaccio v4.2.1 (not sure what version it's not broken for without more research)

**Comment by n4bb12** [comment]
@JeremyTX @seesemichaelj 

I'm not sure what to review as there is no description of intention and changes made.

- Please update the dev dependency to v4, too, if this is intended to be a migration PR.

- Unfortunately this throws an error during login.
```
TypeError: this.auth.issueUIjwt is not a function
```

You can fire up verdaccio with the plugin by running `./run update` or `./run watch` followed by a `./run start` in a bash.

The dev tasks use the `verdaccio.template.yaml` which you can customize locally for testing. Alternatively you can configure your github org via environment variables.

Please make sure the login/logout and local publishing works when requesting a review.

- The PR breaks functionality for verdaccio 3 users. If this is intended to be a migration, please label the PR as such and list the breaking changes.

- I've noticed that incorrect/superseded commands are shown in v4, which were removed in v3. For instance `npm adduser` doesn't make sense in this context.

**Comment by n4bb12** [comment]
@seesemichaelj the other two PRs by @JeremyTX are maked as WIP by JeremyTX himself, so I haven't had a look at those yet.

**Comment by n4bb12** [comment]
@JeremyTX are you still working on this?

**Comment by n4bb12** [comment]
Closing this as it refers to v4 and has been unfinished for a while. Please open a new PR if you intend to complete this.

**Pull request closed without merging**